junit5/src/main/java/org/jboss/weld/junit5/auto/ExcludedBeansExtension.java

```diff
@@ -33,14 +33,26 @@
 public class ExcludedBeansExtension implements Extension {
 
     private Set<Type> excludedBeanTypes;
+    private Set<Class<?>> excludedBeanClasses;
 
-    ExcludedBeansExtension(Set<Type> excludedBeanTypes) {
+    ExcludedBeansExtension(Set<Type> excludedBeanTypes, Set<Class<?>> excludedBeanClasses) {
         this.excludedBeanTypes = excludedBeanTypes;
+        this.excludedBeanClasses = excludedBeanClasses;
     }
 
     <T> void excludeBeans(@Observes @WithAnnotations({Scope.class, NormalScope.class}) ProcessAnnotatedType<T> pat) {
-        if (excludedBeanTypes.contains(pat.getAnnotatedType().getBaseType())) {
+
+        if (excludedBeanClasses.contains(pat.getAnnotatedType().getJavaClass())) {
```

Comment:
Actually not sure from the top of my head, what does `getJavaClass()` return here and what would `getBaseType()` give you?
I think one of these would give you funny results for synthetic ATs and i think it was `getJavaClass()`. I'll have to try tomorrow (not around my PC ATM).

Reply:
They may be interchangeable. The [CDI Spec](http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html) says nothing about it. The reason I used it is because to get the "Java" version of a field, method or constructor you use `getJavaMember()`; seems that's what they are providing to map back to Java. I used `getJavaClass` because we're testing for explicit Java classes here now.

Reply:
Experimenting is giving me same results for both method in various cases. For AT it seems to be equal, so we can use whichever here.